enclave/core/sgx/arena.c

```diff
@@ -20,6 +19,21 @@ static const size_t _max_capacity = 1 << 30;
 void* oe_allocate_arena(size_t capacity);
 void oe_deallocate_arena(void* buffer);
 
+static void _zero_initialize_arena(void)
+{
+    oe_sgx_td_t* td = oe_sgx_get_td();
+    memset(&td->arena, 0, sizeof(td->arena));
+}
+
+static oe_shared_memory_arena_t* _get_arena()
+{
+    static oe_once_t _once = OE_ONCE_INITIALIZER;
+
+    oe_once(&_once, _zero_initialize_arena);
```

Comment:
@jxyang We want the arena to be initialized every time an enclave thread is created, which currently means every time an ecall is made. `oe_once` means that  the action is executed only once; during the lifetime of the enclave.

As written oe_once will get executed for only one of the arenas; not for each arena. If an enclave ha 16 TCS, we want all the arenas to be initialized; not just one.

Note: We are not getting any runtime errors because, oe_teardown_arena zeroes out the memory, effectively doing a `_zero_initialize_arena`.

Reply:
@mikbras The use of `oe_once` to initialize per-thread arena introduces a bug. See my comment above.

Reply:
Oh yikes! You are right. I missed that. Let me rethink this.

Reply:
Okay, I backed out that change. I turns out that the TD is already zero-initialized by **td_init()** so there is no need to zero-initialize the arena.